### PR TITLE
TRUNK-5382: Extra utility method for excluding retired concepts while fetching setmembers

### PR DESCRIPTION
**Pull request by PKatGITHUB**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5382

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you add a unit test for this?

**Comment by samuelmale** [comment]
@PKatGITHUB are you still working on this?

**Comment by PKatGITHUB** [comment]
Hey sorry! I will be solving this by tonight.

**Comment by PKatGITHUB** [comment]
@dkayiwa  I have written the tests. 👍

File: api/src/main/java/org/openmrs/Concept.java [hunk 1]
```diff
@@ -1556,6 +1556,22 @@ public void setId(Integer id) {
 		}
 		return Collections.unmodifiableList(conceptMembers);
 	}
+
+	/**
+	 * @param includeRetired true/false whether to also include the retired concepts
+	 * @return List&lt;Concept&gt; the Concepts that are members of this Concept's set
+	 * @should return the same as getSetMembers() if includeRetired is true
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Concept.java]: In the past, we use these should annotations to generate the corresponding unit tests. If you are not using them for such, we can get rid of them.

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/Concept.java]: Can you we also add a @since annotation?

**Comment by PKatGITHUB** [comment]
@dkayiwa done 👍

**Comment by PKatGITHUB** [comment]
Hey @dkayiwa the build fails at some different Tests (Encounter ones). Can you please have a look at what the problem is??

**Comment by dkayiwa** [comment]
Did you figure this out?

**Comment by PKatGITHUB** [comment]
> Hey @dkayiwa the build fails at some different Tests (Encounter ones). Can you please have a look at what the problem is??

I am not getting why Encounter tests are failing. I have made no changes in that part of code :/ ?

**Comment by PKatGITHUB** [comment]
Can you please help?
@dkayiwa

File: api/src/main/java/org/openmrs/Concept.java [hunk 2]
```diff
@@ -1556,6 +1556,20 @@ public void setId(Integer id) {
 		}
 		return Collections.unmodifiableList(conceptMembers);
 	}
+
+	/**
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Concept.java]: Can you add a description to this method?

File: api/src/main/java/org/openmrs/Concept.java [hunk 3]
```diff
@@ -1556,6 +1556,20 @@ public void setId(Integer id) {
...
+	 * @param includeRetired true/false whether to also include the retired concepts
+	 * @since 2.2
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Concept.java]: This will be 2.3 since 2.2 is already branched off

**Comment by dkayiwa** [comment]
Ok i have run travis again.

**Comment by PKatGITHUB** [comment]
@dkayiwa are we good to go?

**Comment by dkayiwa** [comment]
@PKatGITHUB can you look at other method descriptions to get a feel of how they look like?

**Comment by PKatGITHUB** [comment]
Please review @dkayiwa.
And if you still find anything missing then please point it out where I am mistaken.

File: api/src/main/java/org/openmrs/Concept.java [hunk 4]
```diff
@@ -1556,6 +1556,23 @@ public void setId(Integer id) {
 		}
 		return Collections.unmodifiableList(conceptMembers);
 	}
+
+	/**
+	 * If includeRetired is true, then the returned object is the list of all the concept
+	 * set members of current concept, else retired concept set members are excluded.
+	 *
+	 * @param includeRetired true/false whether to also include/exclude the retired concepts
+	 * @since 2.3
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Concept.java:1565]:
```suggestion
	 * @since 2.5
```

**Pull request merged**